packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -320,37 +295,43 @@ export async function processNodeManifest(
  * and then removes them from the store.
  * Manifest files are added via the public unstable_createNodeManifest action in sourceNodes
  */
-export async function processNodeManifests(): Promise<void> {
+export async function processNodeManifests(): Promise<Record<string, string>> {
+  const startTime = Date.now()
   const { nodeManifests } = store.getState()
 
   const totalManifests = nodeManifests.length
 
   if (totalManifests === 0) {
-    return
+    return {}
   }
 
-  const processedManifests = await Promise.all(
-    nodeManifests.map(manifest => processNodeManifest(manifest))
-  )
-
   let totalProcessedManifests = 0
   let totalFailedManifests = 0
-
-  processedManifests.forEach(manifest => {
-    if (manifest) {
-      totalProcessedManifests++
-    } else {
-      totalFailedManifests++
-    }
-  })
+  const nodeManifestPagePathMap = {}
+
+  await Promise.all(
+    nodeManifests.map(async manifest => {
+      const processedManifest = await processNodeManifest(manifest)
+
+      if (processedManifest && processedManifest?.page?.path) {
+        nodeManifestPagePathMap[processedManifest.page.path] =
+          manifest.manifestId
+        totalProcessedManifests++
+      } else {
+        totalFailedManifests++
```

Comment:
I think this else doesn't make sense anymore, a manifest with no page path didn't fail, there's just no page created for that node.

Reply:
Just to clarify, you mean that there's no point in counting the manifests that failed/don't have a path, we only care about the successful ones now.

Reply:
I mean how it was before was how it should be 👍 no page is a successful manifest but we still want to count failed ones.